### ISSUE TITLE
User Interface sketches (WIP)

### ISSUE THREAD

**Issue by AriHrannar** [veritus/veritus-web#39]
I apologize in advance for my shitty sketches using Gravit :)

# Cases
![cases](https://user-images.githubusercontent.com/6696198/27005772-c244dca6-4e25-11e7-929e-368e23a5bd9d.png)

## What to see on each case
1. Name, who put it forward, date
2. Related promises (name, who made them, fullfillment, match, verified by admin that the they match)
3. Link to parliament, parliament session and althingi
4. Tags

# Promise (maybe not really helpful? More helpful to see it under politician / party)
![promises](https://user-images.githubusercontent.com/6696198/27005572-80099696-4e21-11e7-9561-9d7ae44b6474.png)

## What to see on each promise
1. Name, who put it forward, date
2. Description (we have short and long description, long maybe in a dropdown?)
3. Related cases (name, who made them, parliament session, match, verified by admin that the they match)
4. Tags

# Party
![parties](https://user-images.githubusercontent.com/6696198/27005718-6e962eb2-4e24-11e7-94a0-af9fd2311069.png)

## What to see on each party
1. Name
2. Description 
3. Related promises (name, date, fullfillment, tags)
4. Politicians sorted by district (with % score, maybe sorted?)

# Politician
![politicians](https://user-images.githubusercontent.com/6696198/27005775-c5afae02-4e25-11e7-9c6b-6e4e618ec718.png)

## What to see on each politician
1. Name, score, district
2. Description 
3. Related promises (name, date,fulfillment, tags)
4. Related cases (name, parliament session, status)

# Parliament session
![parliamentsession](https://user-images.githubusercontent.com/6696198/27005816-e2eb0560-4e26-11e7-898d-e3535f604dbd.png)

## What to see on each parliament session
1. Name, to and from dates
2. Cases (name, who made them,  status, tags)

# Districts
![districts](https://user-images.githubusercontent.com/6696198/27005863-5052c8c2-4e27-11e7-85a6-67656baca23b.png)

## What to see on each district
1. Name
2. Politicians (name, rating)

# Dashboard
![dashboard](https://user-images.githubusercontent.com/6696198/27005949-5105993c-4e29-11e7-8853-a9686fcdc776.png)

# Search
![search](https://user-images.githubusercontent.com/6696198/27006003-9ac7f7da-4e2a-11e7-882c-32ff65c784ac.png)

**Comment by Ragnar-Hardarson** [comment]
I don't think it is shitty. 
It gets the point across and now we can have a discussion before we start developing. 

I'm wondering how important we want to make the tags, from a user perspective. 
Of course it's incredibly important on backend to map cases. 
My purpose for asking is : should the tags be at the bottom of the card? 
If they're really important, then display the top3 in the closed card? 

I also quite like descriptive labels. So for the links to parliament, session and alþingi : group them together with a "discussed on:"

**Comment by AriHrannar** [comment]
I could see the tags being useful at least in the cases to get a general idea of what its about. I dont know if the title is always very descriptive.

Might be nice to have 2-3 in the closed card

**Comment by AriHrannar** [comment]
Not entirely sold on the parliament / parliament session being needed. Its mostly for reference if you want to maybe go there and see some other stuff that is being promises / put on althingi at the same time. The althingi link is needed I think just to map the case to the actual website. Im gonna think on it a bit

**Comment by AriHrannar** [comment]
@Ragnar-Hardarson drafts are ready for now. Thoughts?

When we agree on the drafts, I can create stories on the backend / frontend with what is needed to get them to the stage were they look like the images.

**Comment by Ragnar-Hardarson** [comment]
Awesome awesome awesome work.
Really good to have something up to discuss, I didn't even realise how much I needed something like this.

# Thoughts (in no particular order)
- I'd love to see an icon or abbreviation of the political party next to each Politician.
- In detailed Politician view, I'd like to toggle Promises sorting. Between `Fulfilled` and `Date`
- In `Dashboard` Contrast top politicians/party with worst. (We could be cheeky and call them `honest`/`untrustworthy`
- In `Party` show accumulated promise completeness in `Closed` state
- I've fallen in love with [Mediums](https://medium.com/search?q=something) search site. I'd like to do that instead of the side-by-side search. Since side-by-side doesn't scale to mobile very well

### Random idea. Probably out of scope for the MVP
A "cares most about" feature.
Politicians and Parties will have a "Most vocal on" Tag, where it's the tag that shows most up in their promises

**Comment by AriHrannar** [comment]
> I'd love to see an icon or abbreviation of the political party next to each Politician.

The political party is there at the top of the Politicians image. I thought it might be nice to sort them by party instead of having an icon on each one of them?

> In detailed Politician view, I'd like to toggle Promises sorting. Between Fulfilled and Date

Not entirely sure what you what you mean? Do you want to be able to sort them by fulfilled and date?

> In Dashboard Contrast top politicians/party with worst. (We could be cheeky and call them honest/untrustworthy

I like it

> In Party show accumulated promise completeness in Closed state

Maybe like a slider? "50% of promises fulfilled". That might be kewl

> I've fallen in love with Mediums search site. I'd like to do that instead of the side-by-side search. Since side-by-side doesn't scale to mobile very well

I guess it kinda depends on what the search should accomplish. Like on medium, I dont seem to be able to search for "X that has both tag business and entrepreneurship". But I could see that being useful on politech (search for X that has both tag hospital and healthcare)

The string search is pretty cool however.

**Comment by Ragnar-Hardarson** [comment]
> The political party is there at the top of the Politicians image. I thought it might be nice to sort them by party instead of having an icon on each one of them?

I'd still like to see the icon. 
Both for the case of if there are many politicians in a list, and the user scrolls the Party name out of frame. And if we want to do any other grouping, f.x. rating, search results, voted yes on a case, voted no on a case...etc 

> Not entirely sure what you what you mean? Do you want to be able to sort them by fulfilled and date?

Yes.

> I guess it kinda depends on what the search should accomplish. Like on medium, I dont seem to be able to search for "X that has both tag business and entrepreneurship". But I could see that being useful on politech (search for X that has both tag hospital and healthcare)

Speaking for myself, I rarely use a search that isn't just a single text input field.

But be that as it may. The side by side `Search | Results` will not scale to mobile.
